packages/fuel-gauge/src/storage-test-contract.test.ts

```diff
@@ -36,4 +40,46 @@ describe('StorageTestContract', () => {
     const { value: count } = await contract.functions.counter().get();
     expect(count.toHex()).toEqual(toHex(1337));
   });
+
+  it('can increment counter - using custom inline storage slots', async () => {
+    const provider = new Provider('http://127.0.0.1:4000/graphql');
+    const wallet = await generateTestWallet(provider, [[1_000, NativeAssetId]]);
+    const bytecode = readFileSync(
+      join(__dirname, '../test-projects/storage-test-contract/out/debug/storage-test.bin')
+    );
+    const factory = new ContractFactory(bytecode, abi, wallet);
+    // #region contract-deployment-storage-slots-inline
+    const contract = await factory.deployContract({
+      storageSlots: [
```

Comment:
I had to opt against avoiding duplication of code here for the setup because if I did that the snippets would not be as clear.